Tests/SwiftParserTest/DeclarationTests.swift

```diff
@@ -2934,4 +2934,117 @@ final class DeclarationTests: ParserTestCase {
       ]
     )
   }
+
+  // https://github.com/apple/swift-syntax/issues/2273
+  func testEnumCaseWithGenericParameter() {
+    assertParse(
+      """
+      enum Foo {
+        case foo1️⃣<T>(T)
+      }
+      """,
+      diagnostics: [
+        DiagnosticSpec(
+          locationMarker: "1️⃣",
+          message: "generic signature cannot be declared in enum 'case'."
```

Comment:
AFAIK, none of the other error messages end with a period and neither should this.